### DISCUSSION TITLE
CC2652R does not work as a router

### DISCUSSION THREAD

**Discussion by 8none1** [Koenkk/zigbee2mqtt#9479] [Devices]
Hi folks,

I've searched a lot for my issue, and while I see similar problems reported people have typically closed their issues with "it works now" or the issue or discussion has gone stale with no fixes.
So I'm trying again, and I've done some homework already.

I have a zzh! stick running as my co-ordinator and it's working very well.  I also have two CC2531 routers which are also working well.

I decided that I wanted to upgrade my CC2531 routers to zzh! CC2652R routers instead to give me a bit more headroom on the number of devices my network could handle, as I wanted to add a load of TS0601 TRVs.

I bought a new zzh! stick and flashed the `CC2652R_router_20210128` ZStack 3.x.0 firmware.

I added the new router to the network, but got this error while adding:

```
Failed to configure '0x00124b0022a53b1c', attempt 2 (Error: Bind 0x00124b0022a53b1c/8 genBasic from '0x00124b002281a513/1' failed (AREQ - ZDO - bindRsp after 10000ms) at Timeout._onTimeout (/home/pi/zigbee2mqtt/node_modules/zigbee-herdsman/src/utils/waitress.ts:64:35) at listOnTimeout (internal/timers.js:557:17) at processTimers (internal/timers.js:500:7))
```

I'm running zigbee2mqtt `1.22.0 commit: cf68507e`

The router appears in my device list, but the "last seen" time doesn't update and according to the map no devices have tried to use it as a router.  It's not working.

I have also tried the previous firmware - same problem.

I have also forward ported the patches from the ZStack 3.x.0 to the latest TI `zr_genericapp_CC26X2R1_LAUNCHXL_tirtos_ccs` upstream release, but I can't get that to work either.  I'm not hopeful that waiting for the next firmware release to drop will solve this problem.  It seems to me that there is something wrong when trying to use CC2652R as a router.

I have seen multiple other people hint at the same problem, but I can't find a concrete solution.

Is anyone successfully using a CC2652R (specifically this model) as a router?  If so, that implies that there is a problem with my network configuration.  If not, that might imply there is a bug in the router firmware?

Cheers!

**Comment by 8none1** [comment]
I've also now flashed it with the PTVO firmware, but same deal.  The device appears but nothing connects via it, and the `last seen` time never updates.  I think that there is something stuck in the config though, because the web UI shows the `zigbee model` as the old Z Stack firmware.  I'll try building a custom PTVO firmware next.

Update:  I removed the router, deleted all traces of it from config files.  Reflashed with a tweaked PTVO firmware and added a custom converter.  The router is in network and reporting in.  No devices have chosen to connect via it yet, so I will wait and see.

**Comment by 8none1** [comment]
Update for anyone following along:

The tweaked PTVO firmware is working.  I exported a custom converter from the PTVO tool.  The router is visible and working.

However - I now wonder if it was working correctly all along, and the problem was simply that devices didn't feel the need to move to a new router because they were happy where they were. 

I will reflash with the default Z Stack firmware from Koenkk and re-pair some devices very close to the router.  That should encourage them to pair via the router.

Once I've done this I think I will be able to categorically say whether or not the Z Stack based firmware for the CC2652R is working or not.

**Comment by 8none1** [comment]
If anyone wants to try it, the hex file and the customer converter are here:  https://github.com/8none1/CC2653R/blob/main/whizzyrouter.hex

**Comment by thiscantbeserious** [comment]
I can confirm exactly the same behaviour with all firmwares you mentioned however with the 2652P not R - that's the one with the amplifier. I'm using one each for the coordinator and the router - and only the coordinator works properly. The router itself does behave exactly like you reported no matter with which firmware - it pairs, then fails to bind and no client ever joins them no matter what you do.

**Comment by thiscantbeserious** [comment]
@8none1 Would you mind telling me what you exactly customized on your firmware? 

I tried compiling a "custom" firmware based on the router preset for 2652P  with PTVO without manually changing any manufacturer settings or generating a specific converter for it yet.

With the exact same result of no client ever connecting to it after the spawn of a few hours (they do to a slaesh based router running on CC2652RB with the same antenna, on the same spot, without an Amplifier - I even gave it the same antenna just in case - to no avail).

So I believe those two parts are part of the problem in regards to zigbee2mqtt not handling those repeaters properly.

Proper Identification and Conversion. 

So a bug in zigbee2mqtt or herdsman ... unless you have a specific idea in regards to the firmware-specific things you did of course ...

**Comment by 8none1** [comment]
I didn't actually get around to re-flashing the ZStack 3 firmware and seeing what happened.  I've just checked the map, and there are now zero devices paired via the customised PTVO firmware.  I'm willing to believe that's because they have a better connection to another router, but it's a bit fishy.

I've uploaded my PTVO firmware config ini file and custom [here](https://github.com/8none1/CC2653R).

Here is the ini file directly:
https://github.com/8none1/CC2653R/blob/main/whizzyrouter.ini


FWIW, I agree, I think there is a software bug rather than a firmware bug.

**Reply by thiscantbeserious** [reply]
Either that or its a bug in the SDK that's prevent it from working correctly 

(it tried all combinations from January 2021 SDK starting until the latest one currently available). 

I have absolutely no clue where to look at now. Maybe you have an idea @Koenkk ?

I also tried various things, like starting with a fresh config, latest edge, latest stable - switching the pan ID, switching the coordinator firmware. Re-pairing all devices. Removing my Slaesh based router. Waiting for 24/48 hours. Nothing in the logs indicate that clients failed to use the router - they don't even try - but zigbee2mqtt certainly acknowdleges it as a router.

Nothing ever changed the fact that the router pairs successfully, has an LQI above 100 (20 dBm, less with the default firmware) - connects to the coordinator and the CC2652RB router but doesn't get any client - ever. 

I'm using both a CC2652P as a coordinator and a CC2652RB as a router - which work flawlessly at what they do - the additional CC2652P router simply refuses to work under any circumstance and it doesn't seem like its related to the devices itself because they choose the CC2652RB based router just fine (same location, even switched antennas).

Both CC2652R and CC2652P seem to be affected on current edge version - router only.

**Comment by 8none1** [comment]
I've just taken delivery of a `Sonoff Zigbee 3.0 USB Dongle Plus` so I have created an entirely new network using that as the coordinator and a zzh! (CC2652R) as a router.  I have flashed the Sonoff with both with the latest stable ZStack 3.x.0 firmware and also the development branch.  There is only [one firmware version](https://github.com/Koenkk/Z-Stack-firmware/tree/develop/router/Z-Stack_3.x.0/bin) for the zzh! router, 

On the stable firmware I have got the router to pair with the coordinator, but I cannot get a Xiaomi PIR occupation sensor or a Xiaomi temperature sensor to join the network.

On the develop branch the router tries to join the network and then fails:

```
Zigbee2MQTT:error 2021-11-30 11:17:09: Failed to configure '0x00124b0022a53b1c', attempt 1 (Error: Bind 0x00124b0022a53b1c/8 genBasic from '0x00124b0024c9edd4/1' failed (AREQ - ZDO - bindRsp after 10000ms)
    at Timeout._onTimeout (/opt/zigbee2mqtt/node_modules/zigbee-herdsman/src/utils/waitress.ts:64:35)
    at listOnTimeout (internal/timers.js:557:17)
    at processTimers (internal/timers.js:500:7))
```


Update:  I have just flashed the router firmware on to the Sonoff device and done the "unplug it, and replug it three times" thing to get it to join a network, and it has successfully paired with my original network.  I will monitor to see if any devices actually try to use it as a router or not.

**Comment by 8none1** [comment]
Update on the above:  Two devices (both TS0601 TRVs) have decided to connect via the Sonoff router.  I will continue to monitor the situation over the next few days.

**Comment by 8none1** [comment]
Update:  I've just moved the Sonoff up in to the roof space to act as the main router for upstairs, and devices are choosing to connect to it as a router and so far... things seem to be working.  

The *only* thing I think I did differently was to do the "unplug, plug in three times" thing.

I'm going to retest with another zzh! flashed with the ZStack router firmware and see if that works too.  Will report back.

**Comment by 8none1** [comment]
![image](https://user-images.githubusercontent.com/6552931/144720376-e3f589ec-a5eb-4aa4-ae71-2f5933c0305a.png)

And now devices are connecting via a CC2650R zzh! router.

I can only think that the secret was replugging it three times?

Certainly worth trying.

**Comment by daha76** [comment]
Hello,

I trying to get zzh! CC2652R device working as a router. I flashed latest firmware I found here https://github.com/Koenkk/Z-Stack-firmware/tree/develop/router/Z-Stack_3.x.0/bin. Unfortunately router is not going to join my network where the coordinator is Sonoff. Even if I flashed router firmware zzh! joining network as coordinator - pretty strange.  Never the less from the logs it seems like pairing was successful but at the end I can't see it in the list of paired devices. With CC2531 as router I have no problem.

Is there any progress on this issue? Any tips? 

Zigbee2MQTT version: [1.23.0](https://github.com/Koenkk/zigbee2mqtt/releases/tag/1.23.0) commit: [6ea61bae](https://github.com/Koenkk/zigbee2mqtt/commit/6ea61bae)
Coordinator type: zStack3x0
Coordinator revision: 20210708

Here are log messages:

`Failed to configure 'Coordinator', attempt 1 (Error: Bind 0x00124b0021b776a9/8 genBasic from '0x00124b0021b776a9/1' failed (AREQ - ZDO - bindRsp after 10000ms) at Timeout._onTimeout (/opt/zigbee2mqtt/node_modules/zigbee-herdsman/src/utils/waitress.ts:64:35) at listOnTimeout (internal/timers.js:554:17) at processTimers (internal/timers.js:497:7))
Info MQTT publish: topic 'zigbee2mqtt/CC2531 router', payload '{"last_seen":"2022-02-10T17:51:46+01:00","led":false,"linkquality":51}'
Info Successfully interviewed 'Coordinator', device has successfully been paired
Info Device 'Coordinator' is supported, identified as: Custom devices (DiY) Texas Instruments router (ti.router)
Info MQTT publish: topic 'zigbee2mqtt/bridge/event', payload '{"data":{"definition":{"description":"Texas Instruments router","exposes":[{"access":1,"description":"Link quality (signal strength)","name":"linkquality","property":"linkquality","type":"numeric","unit":"lqi","value_max":255,"value_min":0}],"model":"ti.router","options":[],"supports_ota":false,"vendor":"Custom devices (DiY)"},"friendly_name":"Coordinator","ieee_address":"0x00124b0021b776a9","status":"successful","supported":true},"type":"device_interview"}'
Info Configuring 'Coordinator'
Info MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"message":"interview_successful","meta":{"description":"Texas Instruments router","friendly_name":"Coordinator","model":"ti.router","supported":true,"vendor":"Custom devices (DiY)"},"type":"pairing"}'
Info MQTT publish: topic 'zigbee2mqtt/Coordinator', payload '{"last_seen":"2022-02-10T17:51:52+01:00","linkquality":210}'
Info Successfully configured 'Coordinator'`

Thanks a lot.

**Reply by thiscantbeserious** [reply]
I fixed it by getting Phillips Power Plugs instead. Power-Wise they are quite strong and stable - connect much faster and don't have any issues at all. As a plus those also don't have any external Antennas that get in the way and can be used for non-smart devices too.

CC2652R stays at being my coordinator tought. No issues with that.

**Reply by kovaga** [reply]
> Hello,
> 
> I trying to get zzh! CC2652R device working as a router. I flashed latest firmware I found here https://github.com/Koenkk/Z-Stack-firmware/tree/develop/router/Z-Stack_3.x.0/bin. Unfortunately router is not going to join my network where the coordinator is Sonoff. Even if I flashed router firmware zzh! joining network as coordinator - pretty strange. Never the less from the logs it seems like pairing was successful but at the end I can't see it in the list of paired devices. With CC2531 as router I have no problem.
> 
> Thanks a lot.


I ahve just tried connecting the25652R1 as a router to the network, flashed using the following firmware [CC2652R_router_20221102.zip]
And the behaviour is exactly the same as you described, it joins the netwrok as a "Coordinator", then gets sucesfully configured, 
but never appears in the list of devices.

Zigbee2MQTT version     [1.42.0]
Coordinator type     zStack3x0
Coordinator revision     20240710
